index.js

```diff
@@ -44,48 +45,85 @@ function cleanClock (clock) {
 }
 
 exports.init = function (sbot, config) {
-  const dir = config.path ? path.join(config.path, 'ebt') : null
-  const store = Store(dir, null, toUrlFriendly)
-
-  const ebt = EBT({
-    logging: config.ebt && config.ebt.logging,
-    id: sbot.id,
-    getClock (id, cb) {
-      store.ensure(id, function () {
-        const clock = store.get(id) || {}
-        cleanClock(clock)
-        cb(null, clock)
-      })
-    },
-    setClock (id, clock) {
-      cleanClock(clock, 'non-feed key when saving clock')
-      store.set(id, clock)
-    },
-    getAt (pair, cb) {
-      sbot.getAtSequence([pair.id, pair.sequence], (err, data) => {
-        cb(err, data ? data.value : null)
-      })
-    },
-    append (msg, cb) {
-      sbot.add(msg, (err, msg) => {
-        cb(err && err.fatal ? err : null, msg)
-      })
-    },
-    isFeed: isFeed
-  })
+  const formats = {
+    'classic': require('./formats/classic')
+  }
+
+  const ebts = {}
+  function addEBT(formatName) {
+    const dirName = 'ebt' + (formatName === 'classic' ? '' : formatName)
+    const dir = config.path ? path.join(config.path, dirName) : null
+    const store = Store(dir, null, toUrlFriendly)
+
+    const format = formats[formatName]
+    // EBT expects a function of only feedId so we bind sbot here
+    format.isFeed = (feedId) => format.sbotIsFeed(sbot, feedId)
+
+    const ebt = EBT(Object.assign({
+      logging: config.ebt && config.ebt.logging,
+      id: sbot.id,
+      getClock (id, cb) {
+        store.ensure(id, function () {
+          const clock = store.get(id) || {}
+          cleanClock(clock, format.isFeed)
+          cb(null, clock)
+        })
+      },
+      setClock (id, clock) {
+        cleanClock(clock, format.isFeed)
+        store.set(id, clock)
+      },
+      getAt (pair, cb) {
+        format.getAtSequence(sbot, pair, cb)
+      },
+      append (msgVal, cb) {
+        format.appendMsg(sbot, msgVal, cb)
+      }
+    }, format))
+
+    ebts[formatName] = ebt
+  }
+
+  function getEBT(formatName) {
+    const ebt = ebts[formatName]
+    if (!ebt)
+      throw new Error('Unknown format: ' + formatName)
+
+    return ebt
+  }
+
+  addEBT('classic')
 
   const initialized = DeferredPromise()
 
   sbot.getVectorClock((err, clock) => {
     if (err) console.warn('Failed to getVectorClock in ssb-ebt because:', err)
-    ebt.state.clock = clock || {}
-    ebt.update()
-    initialized.resolve()
+
+    const readies = Object.values(formats).map(f => f.isReady(sbot))
+    Promise.all(readies).then(() => {
+      for (let formatName in ebts) {
+        const format = formats[formatName]
+        const ebt = ebts[formatName]
+
+        validClock = {}
+        for (let k in clock)
+          if (format.isFeed(k))
```

Comment:
Is it possible to just use `ebt.isFeed(k)` here? (Not sure if I understood the `EBT()` constructor correctly or not)